test/samplers/test_samplers.py

```diff
@@ -0,0 +1,202 @@
+import contextlib
+import random
+import re
+
+import pytest
+import torch
+from torchcodec.decoders import FrameBatch, SimpleVideoDecoder
+from torchcodec.samplers import clips_at_random_indices
+
+from ..utils import assert_tensor_equal, NASA_VIDEO
+
+
+@pytest.mark.parametrize("num_indices_between_frames", [1, 5])
+def test_random_sampler(num_indices_between_frames):
+    decoder = SimpleVideoDecoder(NASA_VIDEO.path)
+    num_clips = 2
+    num_frames_per_clip = 3
+
+    clips = clips_at_random_indices(
+        decoder,
+        num_clips=num_clips,
+        num_frames_per_clip=num_frames_per_clip,
+        num_indices_between_frames=num_indices_between_frames,
+    )
+
+    assert isinstance(clips, list)
+    assert len(clips) == num_clips
+    assert all(isinstance(clip, FrameBatch) for clip in clips)
+    expected_clip_data_shape = (
+        num_frames_per_clip,
+        3,
+        NASA_VIDEO.height,
+        NASA_VIDEO.width,
+    )
+    assert all(clip.data.shape == expected_clip_data_shape for clip in clips)
+
+    # Check the num_indices_between_frames parameter by asserting that the
+    # "time" difference between frames in a clip is the same as the "index"
+    # distance.
+    avg_distance_between_frames_seconds = torch.concat(
+        [clip.pts_seconds.diff() for clip in clips]
+    ).mean()
+    assert avg_distance_between_frames_seconds == pytest.approx(
+        num_indices_between_frames / decoder.metadata.average_fps
+    )
+
+
+@pytest.mark.parametrize(
+    "sampling_range_start, sampling_range_end, assert_all_equal",
+    (
+        (10, 11, True),
+        (10, 12, False),
+    ),
+)
+def test_random_sampler_range(
+    sampling_range_start, sampling_range_end, assert_all_equal
+):
+    # Test the sampling_range_start and sampling_range_end parameters by
+    # asserting that all clips are equal if the sampling range is of size 1,
+    # and that they are not all equal if the sampling range is of size 2.
+
+    # Since this has a low but non-zero probability of failing, we hard-code a
+    # seed that works.
+    torch.manual_seed(0)
+
+    decoder = SimpleVideoDecoder(NASA_VIDEO.path)
+
+    clips = clips_at_random_indices(
+        decoder,
+        num_clips=10,
+        num_frames_per_clip=2,
+        sampling_range_start=sampling_range_start,
+        sampling_range_end=sampling_range_end,
+    )
+
+    cm = (
+        contextlib.nullcontext()
+        if assert_all_equal
+        else pytest.raises(AssertionError, match="Tensor-likes are not")
+    )
+    with cm:
```

Comment:
A comment would be useful here. I don't understand the purpose or even function of what this is doing (as a non-python user).